packages/experiments/src/components/SelectedItemsList/ControlledSelectedItemsList.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { IControlledSelectedItemsList, IControlledSelectedItemListProps, BaseSelectedItem } from './SelectedItemsList.types';
+import { copyToClipboard } from '../../utilities/copyToClipboard';
+import { useSelectionIndices } from '../../utilities/useSelectionIndices';
+
+/**
+ * An externally-managed selected item list.
+ */
+export const ControlledSelectedItemsList = <TItem extends BaseSelectedItem>(props: IControlledSelectedItemListProps<TItem>) => {
+  const selection = React.useMemo(() => props.selection, [props.selection]);
+  const selectedIndices = useSelectionIndices(selection);
+  // Synchronize the selection against the items in the selection
+  React.useEffect(() => {
+    selection &&
+      selection.setItems(
+        props.selectedItems,
+        false // shouldClear
+      );
+  });
+
+  const onRemoveItemCallbacks = React.useMemo(
+    () =>
+      // create callbacks ahead of time with memo.
+      // (hooks have to be called in the same order)
+      props.selectedItems.map((item: TItem) =>
+        // do not generate callbacks for items that cannot be removed
+        !props.canRemoveItem || props.canRemoveItem(item) ? () => props.onItemsRemoved([item]) : undefined
+      ),
+    [props.selectedItems, props.canRemoveItem, props.onItemsRemoved]
+  );
+
+  // only used in the imperative handle
+  const copyItemsInSelectionToClipboard = React.useCallback((): void => {
+    if (props.getItemCopyText && selectedIndices.length > 0) {
+      const itemsInSelection = selectedIndices.map(itemIndex => props.selectedItems[itemIndex]);
+      const copyText = props.getItemCopyText(itemsInSelection);
+      copyToClipboard(copyText);
+    }
+  }, [props.selectedItems, props.getItemCopyText, selectedIndices]);
+
+  React.useImperativeHandle(
```

Comment:
Since this one is meant to be controlled should this be moved out? I'm not quite sure if that's the right move.

Reply:
This is the right call, but `copyItemsInSelectionToClipboard` is something that should probably still exist on the Controlled component API we present to users.

This + the above is pushing me towards having a separate "View" component & user-presented "Control" component.